src/Domain/ElementIndex.hpp

```diff
@@ -29,6 +29,10 @@ class SegmentIndex {
  public:
   SegmentIndex() = default;
   SegmentIndex(size_t block_id, const SegmentId& segment_id) noexcept;
+  size_t block_id() const noexcept { return block_id_; }
+  size_t index() const noexcept { return index_; }
+  size_t refinement_level() const noexcept { return refinement_level_; }
+
```

Comment:
with these defined you can use them in the stream operator and it no longer needs to be a friend